terror_error.go

```diff
@@ -122,16 +115,50 @@ func (e *Error) MessageTemplate() string {
 	return e.message
 }
 
+// NewError creates a standard error object.
+func NewError(code int, message string) *Error {
+	return &Error{
+		code:    ErrCode(code),
+		message: message,
+	}
+}
+
+// NewErrorWithText creates a standard error object using text error code.
+func NewErrorWithText(code string, message string) *Error {
+	return &Error{
+		codeText: ErrCodeText(code),
+		message:  message,
+	}
+}
+
+// SetWorkaround sets the workaround for standard error.
+func (e *Error) SetWorkaround(workaround string) *Error {
+	e.Workaround = workaround
+	return e
+}
+
+// SetWorkaround sets the description for standard error.
+func (e *Error) SetDescription(description string) *Error {
+	e.Description = description
+	return e
+}
+
+// SetErrCodeText sets the text error code for standard error.
+func (e *Error) SetErrCodeText(codeText string) *Error {
+	e.codeText = ErrCodeText(codeText)
+	return e
+}
+
 // Error implements error interface.
 func (e *Error) Error() string {
 	describe := e.codeText
 	if len(describe) == 0 {
 		describe = ErrCodeText(strconv.Itoa(int(e.code)))
 	}
-	return fmt.Sprintf("[%s] %s", e.RFCCode(), e.getMsg())
+	return fmt.Sprintf("[%s] %s", e.RFCCode(), e.GetMsg())
```

Comment:
I think we should use the local variable `describe` as the prefix of the error message.